### PR TITLE
llvm: strip executables/libraries/object files

### PR DESCRIPTION
**Pull request by maxim-belkin**

### CODE REVIEW

File: Formula/llvm.rb [hunk 1]
```diff
@@ -158,7 +158,7 @@ class Llvm < Formula
   unless OS.mac?
     depends_on "gcc" # <atomic> is provided by gcc
     depends_on "glibc" => GlibcRequirement.system_version.to_f >= 2.19 ? :optional : :recommended
-    depends_on "binutils" if OS.linux? # needed for gold plugin
+    depends_on "binutils" if OS.linux? # needed for gold plugin and stripping executables/libraries/object files
```
Comment (sjackman) [inline Formula/llvm.rb]: You can get rid of `if OS.linux?`. It's already inside a `unless OS.mac?` block.
The comment could be abbreviated to `# needed for gold and strip`

File: Formula/llvm.rb [hunk 2]
```diff
@@ -326,6 +326,9 @@ def install
     # libgomp.so conflicts with gcc.
     # libunwind.so conflcits with libunwind.
     rm [lib/"libgomp.so", lib/"libunwind.so"] if OS.linux?
+
+    # Strip executables/libraries/object files to reduce their size
+    system "strip", *(Dir[bin/"*", lib/"*"]).select { |f| Pathname.new(f).file? && Pathname.new(f).elf? } unless OS.mac?
```
Comment (sjackman) [inline Formula/llvm.rb]: Change to…
```ruby
.select { |f| f = Pathname.new(f); f.file? && f.elf? }
```
(unless `brew style` barfs on it)

File: Formula/llvm.rb (hunk 1 above)
Comment (maxim-belkin) [inline Formula/llvm.rb]: `gold` is Linux-only. no raspberry-pi, arm, etc.

File: Formula/llvm.rb (hunk 2 above)
Comment (maxim-belkin) [inline Formula/llvm.rb]: will do

Comment (maxim-belkin) [inline Formula/llvm.rb]:
```
== Formula/llvm.rb ==
C:331: 78: Do not use semicolons to terminate expressions.
```

Comment (maxim-belkin) [inline Formula/llvm.rb]: `f = Pathname.new(f) && f.file? && f.elf?` ?

Comment (sjackman) [inline Formula/llvm.rb]: Nah, let's not torture it. What you had was fine. The alternative would be to break it over multiple lines, but I'm okay with your one liner. You could also chain:
```ruby
.map { |f| Pathname.new(f) }.select( |f| f.file? && f.elf? }
```
I don't have a strong preference.

File: Formula/llvm.rb (hunk 1 above)
Comment (sjackman) [inline Formula/llvm.rb]: Raspberry-pi, ARM are also `OS.linux?`. Did you mean `x86_64`?

File: Formula/llvm.rb (hunk 2 above)
Comment (maxim-belkin) [inline Formula/llvm.rb]: assuming that this bug will be fixed and we won't need `.file?` I'll leave my one liner then

Comment (sjackman) [inline Formula/llvm.rb]: :+1:

Comment (sjackman) [inline Formula/llvm.rb]: Would you care to open a PR to fix the bug?

Comment (maxim-belkin) [inline Formula/llvm.rb]: don't know at the moment if I have time for that :)

Comment (sjackman) [inline Formula/llvm.rb]: https://github.com/Linuxbrew/brew/blob/master/Library/Homebrew/extend/pathname.rb#L310
```diff
-read(4) == "\x7fELF"
+file? && read(4) == "\x7fELF"
```
should do it.

**Comment by sjackman** [comment]
Looks good to me. Please squash before merging.

**Comment by maxim-belkin** [comment]
I've initiated the last test before merging this

**Pull request closed without merging**